src/chips/nrf51822/gpio.rs

```diff
@@ -1,17 +1,76 @@
 use core::mem;
 use core::cell::Cell;
 use core::ops::{Index, IndexMut};
-use hil;
-
+use peripheral_interrupts::NvicIdx;
 use common::take_cell::TakeCell;
+use common::VolatileCell;
+use hil;
+use chip;
+use nvic;
 
 use peripheral_registers::{GPIO_BASE, GPIO};
 
+struct GpioteRegisters {
+    pub out0:     VolatileCell<u32>, // 0x0
+    pub out1:     VolatileCell<u32>, // 0x4
+    pub out2:     VolatileCell<u32>, // 0x8
+    pub out3:     VolatileCell<u32>, // 0xC
+    _reserved0:     [VolatileCell<u8>; 0xF0],
+    pub in0:      VolatileCell<u32>, // 0x100
+    pub in1:      VolatileCell<u32>, // 0x104
+    pub in2:      VolatileCell<u32>, // 0x108
+    pub in3:      VolatileCell<u32>, // 0x10C
+    _reserved1:     [VolatileCell<u8>; 0x6C],
+    pub port:     VolatileCell<u32>, // 0x17C,
+    _reserved2:     [VolatileCell<u8>; 0x180],
+    pub inten:    VolatileCell<u32>, // 0x300
+    pub intenset: VolatileCell<u32>, // 0x304
+    pub intenclr: VolatileCell<u32>, // 0x308
+    _reserved3:     [VolatileCell<u8>; 0x204],
+    pub config0:  VolatileCell<u32>, // 0x510
+    pub config1:  VolatileCell<u32>, // 0x514
+    pub config2:  VolatileCell<u32>, // 0x518
+    pub config3:  VolatileCell<u32>, // 0x51C
+}
+
+const GPIOTE_BASE: u32 = 0x40006000;
```

Comment:
this should just be a `usize` (or even better a `*const GPIO` if the compiler will allow it) so the cast below is not required.